redshift_sqlalchemy/dialect.py

```diff
@@ -189,17 +303,106 @@ def set_isolation_level(self, connection, level):
 
         connection.set_isolation_level(level)
 
-    def _get_column_info(self, *args, **kwargs):
+    def _get_column_info(self, col, domains):
         column_info = super(RedshiftDialect, self)._get_column_info(
-            *args,
-            **kwargs
-        )
+            name=col.name, format_type=col.format_type, default=col.default,
+            notnull=col.notnull, domains=domains, enums=[], schema=col.schema)
         if isinstance(column_info['type'], VARCHAR):
             if column_info['type'].length is None:
                 column_info['type'] = NullType()
-
+        if 'info' not in column_info:
+            column_info['info'] = {}
+        if col.encoding and col.encoding != 'none':
+            column_info['info']['encode'] = col.encoding
         return column_info
 
+    @reflection.cache
+    def _get_all_table_and_view_info(self, connection):
+        keyfunc = lambda x: (x.schema, x.relname)
+        result = connection.execute("""
+        SELECT
```

Comment:
The Postgres dialect defines these system queries via straight SQL in text, so I'm following that pattern here. I wouldn't be opposed to migrating to Core query expressions in the future. I'm just getting started with SQLAlchemy, so it's not obvious to me how to refactor these.